src/writers/sink.creator.cpp

```diff
@@ -25,14 +26,35 @@ zarr::SinkCreator::make_data_sinks(
 {
     std::queue<std::string> paths;
 
-    std::string base_dir = base_uri;
-    if (base_uri.starts_with("file://")) {
-        base_dir = base_uri.substr(7);
-    }
-    paths.emplace(base_dir);
+    bool is_s3 = common::is_s3_uri(base_uri);
+    std::string bucket_name;
+    if (is_s3) {
+        auto tokens = common::split_uri(base_uri);
+        CHECK(tokens.size() > 2);
+        bucket_name = tokens.at(2);
```

Comment:
Yeah, this is a lesson I've learned since starting this project. This will take a change to the [StorageProperties](https://github.com/acquire-project/acquire-common/blob/main/acquire-core-libs/src/acquire-device-properties/device/props/storage.h#L43) struct, which I'm going to have to take care of separately.